dali/python/nvidia/dali/plugin/pytorch/__init__.py

```diff
@@ -795,3 +853,266 @@ def __next__(self) -> List[Dict[str, torch.Tensor]]:
     DENSE_TAG: str = "dense"
     SPARSE_LIST_TAG: str = "sparse_list"
     SPARSE_COO_TAG: str = "sparse_coo"
+
+
+class DALIProxy:
+    """
+    Proxy to communicate to send processing requests to a DALI pipeline running on the main loop.
+    This is used by PyTorch data workers to assign some processing to the loaded samples, which can
+    execute in the main process via the GPU.
+
+    Background: As the PyTorch workers run on separate processes, using the GPU directly from each
+    of those is not ideal for performance, due to the usage of several CUDA contexts.
+    """
```

Comment:
An ASCII drawing illustrating the processes and the data flow between them would be very useful here.

Reply:
Wonderful, thank you!